tests/Unit/DataStructures/Tensor/Expressions/Test_EvaluateTimeIndex.cpp

```diff
@@ -0,0 +1,130 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <random>
+
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "DataStructures/Tensor/Symmetry.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+// \brief Test evaluation of tensors where concrete time indices are used for
+// RHS spacetime indices
+//
+// \tparam DataType the type of data being stored in the expression operands
+template <typename DataType, typename Generator>
+void test_rhs(const DataType& used_for_size,
```

Comment:
`not_null` arguments always go first (apply everywhere you're adding code please, multiple in this file)